refactor: fix mypy issues in checkers/d*.py

Fix mypy errors for the following:

```
cve_bin_tool/checkers/domoticz.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/domoticz.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/dhcpcd.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/dhcpcd.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/davfs2.py:16: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/davfs2.py:17: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/darkhttpd.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/darkhttpd.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
```

Helpful resources:

- [Python typing documentation](https://docs.python.org/3/library/typing.html)
- [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
- If you can't find the answer in the docs, googling the error messages above will often help!
- [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md)
- If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:

- cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR. A good potential title for this one is in the title of this issue.
- You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/checkers/darkhttpd.py
+++ b/cve_bin_tool/checkers/darkhttpd.py
@@ -10,11 +10,13 @@ https://www.cvedetails.com/product/112161/Darkhttpd-Project-Darkhttpd.html?vendo
 Note: darkhttpd is not provided on debian and openWRT. Tests use fedora packages only
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class DarkhttpdChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"darkhttpd/([0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("darkhttpd_project", "darkhttpd")]

--- a/cve_bin_tool/checkers/davfs2.py
+++ b/cve_bin_tool/checkers/davfs2.py
@@ -9,11 +9,13 @@ https://www.cvedetails.com/product/5417/Davfs2-Davfs2.html?vendor_id=3075
 https://www.cvedetails.com/product/26180/Werner-Baumann-Davfs2.html?vendor_id=12878
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class Davfs2Checker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"davfs2 ([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("davfs2", "davfs2"), ("werner_baumann", "davfs2")]

--- a/cve_bin_tool/checkers/dhcpcd.py
+++ b/cve_bin_tool/checkers/dhcpcd.py
@@ -10,12 +10,14 @@ https://www.cvedetails.com/product/20668/Roy-Marples-Dhcpcd.html?vendor_id=11378
 https://www.cvedetails.com/product/28439/Dhcpcd-Project-Dhcpcd.html?vendor_id=13642
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class DhcpcdChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"dhcpcd ([0-9]+\.[0-9]+(\.[0-9]+)?)"]
     VENDOR_PRODUCT = [
         ("phystech", "dhcpcd"),

--- a/cve_bin_tool/checkers/domoticz.py
+++ b/cve_bin_tool/checkers/domoticz.py
@@ -10,11 +10,13 @@ https://www.cvedetails.com/product/52466/Domoticz-Domoticz.html?vendor_id=19702
 Note: domoticz is not provided by debian or openWRT. Tests provided are fedora-only.
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class DomoticzChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"domoticz-([0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("domoticz", "domoticz")]

